File: classroom/src/types/events.ts
import { DateTime } from "luxon";

export type EventsData = {
  /** イベントがあるか */
  isEmpty: boolean;

  /** イベントリスト */
  items: EventData[];
};

export type EventData = {
  /** イベント名 */
  title: string;
  /** 全日イベントかどうか */
  isAllDay: boolean;

  /** 開始日時 */
  startAt: DateTime;
  /** 終了日時 */
  endAt: DateTime;
};

File: classroom_flutter/lib/models/events.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'events.freezed.dart';
part 'events.g.dart';

@freezed
class EventList with _$EventList {
  const factory EventList({
    required List<EventData> items,
  }) = _EventList;

  factory EventList.fromJson(Map<String, dynamic> json) =>
      _$EventListFromJson(json);
}

@freezed
class EventData with _$EventData {
  const factory EventData({
    required String title,
    required bool isAllDay,
    required DateTime startAt,
    required DateTime endAt,
  }) = _EventData;

  factory EventData.fromJson(Map<String, dynamic> json) =>
      _$EventDataFromJson(json);
}
